lib/complexity_metric.dart

```diff
@@ -1,7 +1,46 @@
 library complexity_metric;
 
+import 'package:analyzer/error/listener.dart';
+import 'package:complexity_metric/visitor/complexity_metric_flow_visitor.dart';
+import 'package:custom_lint_builder/custom_lint_builder.dart';
+
+/// creates plugin
+PluginBase createPlugin() => _ComplexityMetricLinter();
+
+/// Complexity metric linter
+class _ComplexityMetricLinter extends PluginBase {
+  @override
+  List<LintRule> getLintRules(CustomLintConfigs configs) => [
+        const ComplexityMetric(),
+      ];
+}
+
 /// A Complexity metric.
-class ComplexityMetric {
-  /// Returns [value] plus 1.
-  int addOne(int value) => value + 1;
+class ComplexityMetric extends DartLintRule {
+  static const _code = LintCode(
+    name: 'complexity_metric',
+    problemMessage: 'Check complexity',
+  );
+
+  static const _minValue = 2;
+
+  /// Constructor class
```

Comment:
Please, add more meaningful documentation similar to the following one: 
```suggestion
  /// Creates a new instance of [ComplexityMetric].
```